compiler/codegen/OMRCodeGenerator.cpp

```diff
@@ -569,6 +570,139 @@ OMR::CodeGenerator::setUpForInstructionSelection()
 
   }
 
+/**
+ * TODO: Remove this function once OMR provides a proper API query function.
+ * OMR issue https://github.com/eclipse/omr/issues/842
+ *
+ *
+ * \brief The ILOpCode property `ILProp1::Call` is a property used in a broader sense.
+ * An opCode that bears this property has call-like side effects; and they
+ * may not be real function calls. This query is used to check if an OpCode is a
+ * functional call by excluding non-funciton-call opCodes.
+ *
+ * \return true if the opCode is a function call; returns false otherwise.
+ *
+ */
+bool
+OMR::CodeGenerator::isFunctionCallNode(TR::Node* node)
+   {
+   return node->getOpCodeValue() == TR::icall       ||
+          node->getOpCodeValue() == TR::lcall       ||
+          node->getOpCodeValue() == TR::fcall       ||
+          node->getOpCodeValue() == TR::dcall       ||
+          node->getOpCodeValue() == TR::acall       ||
+          node->getOpCodeValue() == TR::call        ||
+          node->getOpCodeValue() == TR::callIf      ||
+          node->getOpCodeValue() == TR::vcall       ||
+          node->getOpCodeValue() == TR::vcalli      ||
+          node->getOpCodeValue() == TR::iucall      ||
+          node->getOpCodeValue() == TR::lucall      ||
+          node->getOpCodeValue() == TR::icalli      ||
+          node->getOpCodeValue() == TR::iucalli     ||
+          node->getOpCodeValue() == TR::lcalli      ||
+          node->getOpCodeValue() == TR::lucalli     ||
+          node->getOpCodeValue() == TR::fcalli      ||
+          node->getOpCodeValue() == TR::dcalli      ||
+          node->getOpCodeValue() == TR::acalli      ||
+          node->getOpCodeValue() == TR::calli;
+   }
+
+/**
+ * \brief The uncommon-call-constant-node pass is an extra pre-codegen pass right
```

Comment:
I will delete the word `right`. This was true in my first implementation and is no longer so.